examples/experimental/dagster-airlift/dagster_airlift/in_airflow/dagster_operator.py

```diff
@@ -14,97 +15,131 @@
+                f"Triggering run for {repo_location}/{repo_name}/{job_name} with assets {asset_keys}"
+            )
+            response = session.post(
                 f"{dagster_url}/graphql",
-                json={"query": RUNS_QUERY, "variables": {"runId": run_id}},
+                json={
+                    "query": TRIGGER_ASSETS_MUTATION,
+                    "variables": {"executionParams": execution_params},
+                },
                 timeout=3,
             )
-            run_status = response.json()["data"]["runOrError"]["status"]
-            if run_status in ["SUCCESS", "FAILURE", "CANCELED"]:
-                logger.debug(f"Run {run_id} completed with status {run_status}")
-                completed_runs[run_id] = run_status
-    non_successful_runs = [
-        run_id for run_id, status in completed_runs.items() if status != "SUCCESS"
-    ]
-    if non_successful_runs:
-        raise Exception(f"Runs {non_successful_runs} did not complete successfully.")
-    logger.debug("All runs completed successfully.")
-    return None
-
-
-class DagsterOperator(PythonOperator):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs, python_callable=compute_fn)
-
-
-def build_dagster_task(original_task: BaseOperator) -> DagsterOperator:
-    return instantiate_dagster_operator(original_task)
-
-
-def instantiate_dagster_operator(original_task: BaseOperator) -> DagsterOperator:
+            run_id = response.json()["data"]["launchPipelineExecution"]["run"]["id"]
+            logger.debug(f"Launched run {run_id}...")
+            triggered_runs.append(run_id)
+        completed_runs = {}  # key is run_id, value is status
+        while len(completed_runs) < len(triggered_runs):
+            for run_id in triggered_runs:
+                if run_id in completed_runs:
+                    continue
+                response = session.post(
+                    f"{dagster_url}/graphql",
+                    json={"query": RUNS_QUERY, "variables": {"runId": run_id}},
+                    timeout=3,
+                )
+                run_status = response.json()["data"]["runOrError"]["status"]
+                if run_status in ["SUCCESS", "FAILURE", "CANCELED"]:
+                    logger.debug(f"Run {run_id} completed with status {run_status}")
+                    completed_runs[run_id] = run_status
+        non_successful_runs = [
+            run_id for run_id, status in completed_runs.items() if status != "SUCCESS"
+        ]
+        if non_successful_runs:
+            raise Exception(f"Runs {non_successful_runs} did not complete successfully.")
+        logger.debug("All runs completed successfully.")
+        return None
+
+    def execute(self, context: Context) -> Any:
+        # https://github.com/apache/airflow/discussions/24463
+        os.environ["NO_PROXY"] = "*"
+        dag_id = os.environ["AIRFLOW_CTX_DAG_ID"]
+        task_id = os.environ["AIRFLOW_CTX_TASK_ID"]
```

Comment:
pre-existing, but is there a reason why we get this from `os.environ` rather than the `context`?

Reply:
previously we didn't use the context. Fine with getting it from there too.

Reply:
well we should bias for whatever is maximally backwards compatible and reliable

Reply:
if we change it, let's do it in a sep diff. no reason to change what is working now.